### DISCUSSION TITLE
cashback automático por evento

### DISCUSSION THREAD

**Discussion by scoobiii** [bacen/pix-api#507] [General]
https://bate-papo.kyte.site/
O cliente comprou e pagou pela embalagem portanto tem direito ao cashback de $0,80 que ja inclui todas as tachas do psp/pix

#255 estava com dificuldades com operações de centavos, no caso de uma latinha de aluminio, então subi para 0,80 toda pacoteira e ver que bicho dá.

No pagamento via pix como acesso a chave publica para devolver o pix via processos disparados por  "eventos" de pagamento?
----------------------
Meios de pagamento:
Acesse uma plataforma de pagamentos completa com preços simples e pré-pagos 3,99%  + R$ 0,39 por cobrança bem-sucedida realizada em cartão
Tudo de que você precisa para gerenciar pagamentos
Obtenha centenas de atualizações de recursos a cada ano Sem tarifas iniciais, mensais ou ocultas
----------------------
ABUSIVO
--------------------------

Quanto é esta mesma operação via pix?
Quem são os PSP´s que trabalham com centavos?
Fornecem api que identifica a chave pagadora e
Permite devolução cashback por 'eventos' diversos de pagamentos automático de forma simplificada preferencialmente no & low code?

Até aqui bz porque o cliente comprou e pagou então tenho o crédito do cashback na minha carteira digital, o valor da embalagem mas todo custeio.

Cada embalagem já possui qr code e código de barras então devolvo o cashback uma vez que faço a leitura e reconhecimento da embalagem, previamente, então o crédito que está em meu poder a ser transferido 

Então é dúvida como devolver o cashback para o cliente ?
Preciso de uma carteira para gerir isto e que dispare o pagamento automático receba um evento de pagamento/devolução vinda de um leitor que identifica a embalagem previamente reconhecida/cadastrada
https://github.com/scoobiii/mundrundo5.0/blob/master/app-debug.apk 

neste caso treinada para atribuir valor genérico a pet, mas então treino para reconhecer embalagens únicas e atribuir valor de pagamento por categoria por evento, do tipo, "reconheceu pagou" para pet, vidro, ....tudo igual. todo tipo de embalagem faz o mesmo estrago no aterro.

Então eu preciso ter a chave pública do cliente e linkar o app/leitor a carteira para disparar a devolução ao cliente via chave pix usado no pagamento uma vez que o leitor associa o qr code código de barras a embalagem pois só recebe este cashbak que comprou comigo ou com as lojas cadastradas na campanha, claro, sempre o cliente consciente pagando pela economia circular

Quem é o PSP que recebe o pagamento em pix e como acesso a chave pix publica do cliente que efetuou o pagamento (cnpj, cpf, telefone, @, chave genérica) para devolver o casback via pix?

O foco é priorizar pix pela simplicidade e custo, sem usar o meio de pagamento qr code (de novo) o que encarece ainda mais a operação mas
Se o pagamento for por outros meios de pagamento (cartão/dinheiro) idem idem como identificar o cliente para devolver o casback de forma direta ou o jeito o esquecer cartão como esta fazendo a visa

https://www1.folha.uol.com.br/mercado/2021/11/visa-responde-critica-da-amazon-sobre-taxa-alta-do-cartao-de-credito.shtml
?

pontos? é legal, mas é mais um imbróglio para procrastinar a economia circular

quem nunca ouviu falar 'sustentabilidade para as gerações futuras' um jeito preguiçoso de empurrar com a barriga.

"Mais de 700 toneladas de lixo são retiradas da orla de Praia Grande no carnaval
Segundo a prefeitura, 748 mil kg de lixo foram recolhidos."

https://drive.google.com/drive/folders/12mkFFWWNKTH3AJG101ns_n90Hwa5tvbV?usp=sharing

valew.

**Comment by rubenskuhl** [comment]
Eu acho que o mais prático não é fazer o envio por chave Pix, mas por devolução parcial do Pix. Imagine que o produto tenha um QR-Code Pix para pagamento, do valor que seja o produto (ex: R$5). Aí o cliente faz o pagamento e você recebe o valor na sua conta. 

Aí na ponta da reciclagem, a embalagem aparece. Aí você lê o mesmo QR-Code e com ele,identificada qual Pix e faz devolução de 80 centavos. Com isso, você não precisa ter nenhum cadastro de chaves Pix, a devolução vai ser feita para a conta que fez o pagamento, mesmo você não sabendo que conta é esse ou sequer se tem uma chave Pix associada. Aliás, nesse modelo o comprador pode nem ter chave Pix cadastrada. 

Em termos de economicidade, isso tem uma vantagem que é de que em vários PSPs não há custo adicional de devolução. Então o que quer que tenha custado o recebimento, foi a única taxa que se paga na transação. Um PSP onde isso acontece é a Gerencianet; um PSP onde isso não é assim, e portanto desaconselhado para esse modelo, é o BB. 

Um limitador, que precisa ficar claro para o comprador, é que uma devolução de Pix só pode ser feita em até 90 dias. Então mesmo que o produto tenha uma validade superior a 90 dias, ou que a lata tenha ficado na casa do cliente muito tempo após o consumo, esse "rebate" só vale por 90 dias. 

Mas as vantagens operacionais disso são tantas que eu acho que vale a pena.

**Reply by scoobiii** [reply]
legal. show.
o consumo esta associado a responsabilidade de forma que é preciso saber o comprador por que ele pode transferir o pix ao terceiro responsável para que o resíduo faça a rota correta. Ou seja, ele não pode jogar a lata na rua, praia ou qualquer outro lugar. Um garrafa de uique pode durar mais de 90 dias, então é o plm complementado pelo rlm, ciclo de vida do resíduo.
90 dias é uma restrição ou estimulo?

**Reply by rubenskuhl** [reply]
> legal. show. o consumo esta associado a responsabilidade de forma que é preciso saber o comprador por que ele pode transferir o pix ao terceiro responsável para que o resíduo faça a rota correta. Ou seja, ele não pode jogar a lata na rua, praia ou qualquer outro lugar. Um garrafa de uique pode durar mais de 90 dias, então é o plm complementado pelo rlm, ciclo de vida do resíduo. 90 dias é uma restrição ou estimulo?

Como a sua cadeia de valor vai encarar eu não sei... fato é que é uma restrição imposta pelo BC para a devolução de Pix. 
Só quem pode garantir que a rota do resíduo foi correta é a sua equipe de reciclagem. É ela que só pode passar para devolução um resíduo que tenha seguido caminho apropriado, e não pegar da rua e beneficiar indevidamente quem não fez direitinho. 

Devolução só é feita para o pagador, não tem como ser transferida... mas se o cara quiser transferir dinheiro depois que recebeu da devolução, o dinheiro é dele para doar a quem ele quiser.

**Reply by scoobiii** [reply]
massa.
criamos o plimm nft e moeda
paridade == eth
ai portamos para todas moedas

economia circular, blochain autonoma em relação ao bc

monetizada, no ato da compra.

https://opensea.io/collection/plimm
